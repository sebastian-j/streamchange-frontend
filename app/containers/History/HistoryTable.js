import React, { useState } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import styled from 'styled-components';
import ArrowUpIcon from './arrowUpIcon';
import HistoryItem from './HistoryItem';
import './style.css';

const HeaderButton = styled.button`
  background: none;
  border: none;
  color: ${props => props.theme.staticTextColor};
  padding: 14px 10px 14px 16px;
  outline: none;
  width: 100%;
`;

const Table = styled.table`
  display: table;
  overflow-y: auto;
  min-width: 1052px;
  width: 100%;
`;

const Thead = styled.thead`
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  cursor: pointer;
  user-select: none;
`;

const HistoryTable = props => {
  const [sort, setSort] = useState('createdAtDESC');
  const handleSortChange = event => {
    const { target } = event;
    let value = target.id;
    if (value === sort) {
      value += 'DESC';
    }
    setSort(value);
  };

  return (
    <Table>
      <Thead>
        <tr>
          <td />
          <td>
            <HeaderButton
              id="displayName"
              onClick={handleSortChange}
              type="button"
            >
              Nazwa
              <ArrowUpIcon
                className={clsx(
                  'sort-icon',
                  sort === 'displayName' && 'active',
                  sort === 'displayNameDESC' && [
                    'active',
                    'icon-direction-desc',
                  ],
                )}
              />
            </HeaderButton>
          </td>
          <td>
            <HeaderButton id="prize" onClick={handleSortChange} type="button">
              Nagroda
              <ArrowUpIcon
                className={clsx(
                  'sort-icon',
                  sort === 'prize' && 'active',
                  sort === 'prizeDESC' && ['active', 'icon-direction-desc'],
                )}
              />
            </HeaderButton>
          </td>
          <td>
            <HeaderButton id="message" onClick={handleSortChange} type="button">
              Wiadomość
              <ArrowUpIcon
                className={clsx(
                  'sort-icon',
                  sort === 'message' && 'active',
                  sort === 'messageDESC' && ['active', 'icon-direction-desc'],
                )}
              />
            </HeaderButton>
          </td>
          <td>
            <HeaderButton
              id="createdAt"
              onClick={handleSortChange}
              type="button"
            >
              Data i godzina
              <ArrowUpIcon
                className={clsx(
                  'sort-icon',
                  sort === 'createdAt' && 'active',
                  sort === 'createdAtDESC' && ['active', 'icon-direction-desc'],
                )}
              />
            </HeaderButton>
          </td>
        </tr>
      </Thead>
      <tbody>
        {props.items.map(item => (
          <HistoryItem
            channelId={item.channelId}
            imageUrl={item.imageUrl}
            displayName={item.displayName}
            prize={item.prize}
            message={item.message}
            createdAt={item.createdAt}
          />
        ))}
      </tbody>
    </Table>
  );
};

HistoryTable.propTypes = {
  items: PropTypes.array,
};

export default HistoryTable;
