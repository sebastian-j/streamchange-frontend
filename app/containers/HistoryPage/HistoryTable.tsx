import React, { useState } from 'react';
import clsx from 'clsx';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import ArrowUpIcon from './components/arrowUpIcon';
import HistoryItem from './HistoryItem';
import { Table } from './components/Table';
import { TableHeaderButton } from './components/TableHeaderButton';
import { Thead } from './components/Thead';
import { HistoryItem as HItem } from './types';

interface Props {
  items: HItem[];
}

const HistoryTable = ({ items }: Props) => {
  const [sort, setSort] = useState<string>('createdAtDESC');
  const handleSortChange = (event) => {
    const { target } = event;
    let value = target.id;
    if (value === sort) {
      value += 'DESC';
    }
    setSort(value);
  };

  return (
    <Table>
      <Thead>
        <tr>
          <td />
          <td>
            <TableHeaderButton
              id="displayName"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.nameHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'displayName' && 'active',
                  sort === 'displayNameDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="prize"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.prizeHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'prize' && 'active',
                  sort === 'prizeDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="message"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.messageHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'message' && 'active',
                  sort === 'messageDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="createdAt"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.dateHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'createdAt' && 'active',
                  sort === 'createdAtDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
        </tr>
      </Thead>
      <tbody>
        {items.map((item) => (
          <HistoryItem
            key={item.createdAt}
            channelId={item.channelId}
            imageUrl={item.imageUrl}
            displayName={item.displayName}
            prize={item.prize}
            message={item.message}
            createdAt={item.createdAt}
          />
        ))}
      </tbody>
    </Table>
  );
};

export default HistoryTable;
