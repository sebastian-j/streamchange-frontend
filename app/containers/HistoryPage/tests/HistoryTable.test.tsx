import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import HistoryTable from '../HistoryTable';

const shallowRenderer = createRenderer();

describe('<HistoryTable />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <HistoryTable
        items={[
          {
            channelId: 'id',
            displayName: 'name',
            createdAt: '2019-12-24T07:27:56.273Z',
          },
        ]}
      />,
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
