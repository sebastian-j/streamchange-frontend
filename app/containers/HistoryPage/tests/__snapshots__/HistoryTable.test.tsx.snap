// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryTable /> should render and match the snapshot 1`] = `
<styled.table>
  <styled.thead>
    <tr>
      <td />
      <td>
        <styled.button
          id="displayName"
          onClick={[Function]}
          type="button"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Channel name"
            id="streamchange.containers.History.header.name"
          />
          <ArrowUpIcon
            className=""
          />
        </styled.button>
      </td>
      <td>
        <styled.button
          id="prize"
          onClick={[Function]}
          type="button"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Prize"
            id="streamchange.containers.History.header.prize"
          />
          <ArrowUpIcon
            className=""
          />
        </styled.button>
      </td>
      <td>
        <styled.button
          id="message"
          onClick={[Function]}
          type="button"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Message"
            id="streamchange.containers.History.header.message"
          />
          <ArrowUpIcon
            className=""
          />
        </styled.button>
      </td>
      <td>
        <styled.button
          id="createdAt"
          onClick={[Function]}
          type="button"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Date and time"
            id="streamchange.containers.History.header.date"
          />
          <ArrowUpIcon
            className="active descending"
          />
        </styled.button>
      </td>
    </tr>
  </styled.thead>
  <tbody>
    <HistoryItem
      channelId="id"
      createdAt="2019-12-24T07:27:56.273Z"
      displayName="name"
      imageUrl="url"
      message="test message"
      prize="trophy"
    />
  </tbody>
</styled.table>
`;
